content/docs/studio/user-guide/connect-custom-gitlab-server.md

```diff
@@ -0,0 +1,50 @@
+# Custom GitLab Server Connection
+
+If your team’s Git repositories are on a self-hosted GitLab server, you can set
+up a connection to this server such that all your team members can connect to
+the Git repositories on this server. Refer to the
+[GitLab docs](https://about.gitlab.com/install/) for more details about
+self-hosted GitLab servers.
+
+To connect to your self-hosted GitLab server, you will need the following:
+
+- A URL for your GitLab server. The URL should either be publicly accessible, or
+  in the same private network as Studio if Studio is deployed on-premises.
+- A personal access token with the following roles: `api`, `read_user`,
+  `read_repository`. If you do not have a personal access token yet, you can
+  create one at
+
+  `\<your GitLab server URL>/-/profile/personal_access_tokens?name=Iterative+Studio+Access+token&scopes=api,read_user,read_repository`
```

Comment:
Maybe link to GitLab docs that refer to this instead of providing a URL (that could change in the future)

Reply:
This URL is used in our code as well. So, if/when this changes, we will have to update Studio anyway.